sphinx/texinputs/sphinx.sty

```diff
@@ -1049,17 +1072,28 @@
        \vskip\spx@abovecaptionskip
        \def\sphinxVerbatim@Before
            {\sphinxVerbatim@Title\nointerlineskip
-            \kern\dimexpr-\dp\strutbox+\sphinxbelowcaptionspace\relax}%
+            \kern\dimexpr-\dp\strutbox+\sphinxbelowcaptionspace
+                 % if no frame (code-blocks inside table cells), remove
+                 % the "verbatimsep" whitespace from the top (better visually)
+                 \ifspx@opt@verbatimwithframe\else-\sphinxverbatimsep\fi
```

Comment:
This removal of some vertical space in case the literal-block is in a table cell (so Sphinx uses no framing but still uses `\sphinxverbatimsep` whitespace on the 4 sides) is unrelated to handling caption package, but it improves the PDF, and it was occasion to do so.